_test/map30.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+import "strings"
+
+func f(s string) string { return "hello " + s }
+
+var methods = map[string]func(string) string{"f": f}
+
+func main() {
+	methods["g"] = strings.ToUpper
+	println(methods["f"]("test"))
+	println(methods["g"]("test"))
+}
+
+// Output:
+// hello test
+// TEST
```

Comment:
I expanded the test a little bit to cover the 4 cases of the matrix (interp func, runtime func) X (inserted at initialization, inserted after), and I found one case that still does not work (runtime func inserted at initialization):
```
% yaegi run _test/map30.go 
_test/map30.go:9:15: panic
run: reflect.Value.SetMapIndex: value of type func(string) string is not assignable to type *interp.node
```
```suggestion
package main

import "strings"

func f(s string) string { return "hello " + s }

func g(s string) string { return "hi " + s }

var methods = map[string]func(string) string{
	"f": f,
	"h": strings.ToLower,
}

func main() {
	methods["i"] = strings.ToUpper
	methods["g"] = g
	println(methods["f"]("test"))
	println(methods["g"]("test"))
	println(methods["h"]("TEST"))
	println(methods["i"]("test"))
}

// Output:
// hello test
// hi test
// test
// TEST
```